tower-buffer/src/worker.rs

```diff
@@ -0,0 +1,215 @@
+use error::ServiceError;
+use futures::future::Executor;
+use futures::{Async, Future, Poll, Stream};
+use message::Message;
+use std::sync::{Arc, Mutex};
+use tokio_sync::mpsc;
+use tower_service::Service;
+
+/// Task that handles processing the buffer. This type should not be used
+/// directly, instead `Buffer` requires an `Executor` that can accept this task.
```

Comment:
Maybe add a note that this is only `pub` because we're sealing over it? Specifically, the type is _not_ actually exported by `tower-buffer`.